cli_tools/gce_vm_image_import/main.go

```diff
@@ -141,10 +148,32 @@ func validateAndParseFlags() error {
 			return err
 		}
 	}
+
 	return nil
 }
 
-//Returns main workflow and translate workflow paths (if any)
+// Validate source file is not a compression file by checking file header.
+func validateSourceFile(storageClient commondomain.StorageClientInterface) error {
+	if sourceFile == nil {
+		return nil
+	}
+
+	rc, err := storageClient.GetObjectReader(*sourceBucketName, *sourceObjectName)
+	if err != nil {
+		return fmt.Errorf("readFile: unable to open file from bucket %q, file %q: %v", *sourceBucketName, *sourceObjectName, err)
+	}
+	defer rc.Close()
```

Comment:
This will never be executed in case of an error. Defer calls a method at the end of this method and should be moved before if statement above

Reply:
You dont need to close if GetObjectReader returns an error, you should be able to expect it to not create the client